internal/controller/factory/statefulset.go

```diff
@@ -214,10 +222,43 @@ func generateEtcdCommand(cluster *etcdaenixiov1alpha1.EtcdCluster) []string {
 		"--peer-auto-tls",
 		fmt.Sprintf("--advertise-client-urls=http://$(POD_NAME).%s.$(POD_NAMESPACE).svc:2379", cluster.Name),
 	}
+}
+
+func generateEtcdArgs(cluster *etcdaenixiov1alpha1.EtcdCluster) []string {
+	args := generateBaseEtcdArgs(cluster)
 
 	for name, value := range cluster.Spec.PodSpec.ExtraArgs {
-		command = append(command, fmt.Sprintf("--%s=%s", name, value))
+		flag := "--" + name
+		if len(value) == 0 {
+			args = append(args, flag)
+
+			continue
+		}
+
+		args = append(args, fmt.Sprintf("%s=%s", flag, value))
+	}
+
+	return args
+}
+
+func ValidatePodExtraArgs(cluster *etcdaenixiov1alpha1.EtcdCluster) error {
+	if len(cluster.Spec.PodSpec.ExtraArgs) == 0 {
+		return nil
+	}
+
+	baseArgs := argsFromSliceToMap(generateBaseEtcdArgs(cluster))
+
+	for name := range cluster.Spec.PodSpec.ExtraArgs {
+		if strings.HasPrefix(name, "-") {
+			return fmt.Errorf("the extra argument shoudn't have dashes, flag: %s", name)
```

Comment:
Extra arg should not start from dash and have trailing dash, but it can contain dashes in the middle.

Reply:
fixed.